Vagrantfile

```diff
@@ -0,0 +1,88 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# Read the Kata Containers CI job from the CI_JOB environment variable.
+job = ENV['CI_JOB'] || ""
+guest_home_dir = '/home/vagrant'
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+
+  # By default vagrant sync the current directory. Let's disabled it because the directory
+  # will be synced later to the proper destination.
+  config.vm.synced_folder ".", "/vagrant", disabled:true
+  config.vm.synced_folder ".", "#{guest_home_dir}/go/src/github.com/kata-containers/tests", type:"rsync"
+  config.vm.synced_folder "#{ENV['GOPATH']}/src/github.com/kata-containers/kata-containers",
```

Comment:
Could you add a test that `GOPATH` is set and fail gracefully otherwise? I'm also confused because of the `export` on line 30. Does `Vagrant.configure` run _after_ the "shared provision script"?

Reply:
I will add a trigger to check the value of `GOPATH` before the `up` and `ssh` actions, on its absense it will abort the process. The execution message will be like:
```
Bringing machine 'fedora' up with 'libvirt' provider...
==> fedora: Running action triggers before up ...
==> fedora: Running trigger...
==> fedora: You must export the GOPATH environment variable
==> fedora: Vagrant has been configured to abort. Terminating now...
```
@c3d do you think it is sufficient?

Reply:
Just update this PR with those changes.